wandb/sdk/artifacts/artifact.py

```diff
@@ -1647,12 +1649,16 @@ def download(
 
         Raises:
             ArtifactNotLoggedError: If the artifact is not logged.
+            RuntimeError: If the artifact is attempted to be downloaded in offline mode.
         """
         self._ensure_logged("download")
 
         root = FilePathStr(str(root or self._default_root()))
         self._add_download_root(root)
 
+        if env.is_offline() or self._settings._offline():
```

Comment:
Instead of assigning to a `_settings` member, I think you can look at the `wandb.settings` value here, similar to how the existence of a run is checked when logging

Reply:
Can you suggest the fix? I am having a hard time pinning down the exact ways to check for offline. The codebase is quite inconsistent or not nearly as proactive in doing so.

Reply:
After discussion we're settling on `util._is_offline()`